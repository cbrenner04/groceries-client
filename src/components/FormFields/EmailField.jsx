import React from 'react';
import PropTypes from 'prop-types';

const EmailField = ({
  name,
  label,
  value,
  handleChange,
}) => (
  <div className="form-group">
    <label htmlFor={name}>{label}</label>
    <input
      id={name}
      type="email"
      name={name}
      className="form-control"
      value={value}
      onChange={handleChange}
      placeholder="jane.smith@example.com"
    />
  </div>
);

EmailField.propTypes = {
  name: PropTypes.string,
  label: PropTypes.string,
  value: PropTypes.string.isRequired,
  handleChange: PropTypes.func.isRequired,
};

EmailField.defaultProps = {
  name: 'email',
  label: 'Email',
};

export default EmailField;
