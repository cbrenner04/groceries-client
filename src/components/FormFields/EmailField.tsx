import React, { type ChangeEvent } from 'react';
import { Form } from 'react-bootstrap';

export interface IEmailFieldProps {
  value: string;
  handleChange: ({ target: { value } }: ChangeEvent<HTMLInputElement>) => void;
  name?: string;
  label?: string;
  placeholder?: string;
}

const EmailField: React.FC<IEmailFieldProps> = (props): React.JSX.Element => (
  <Form.Group controlId={props.name ?? 'email'} className="mb-3">
    <Form.Label>{props.label ?? 'Email'}</Form.Label>
    <Form.Control
      type="email"
      value={props.value}
      onChange={props.handleChange}
      placeholder={props.placeholder ?? 'jane.smith@example.com'}
      name={props.name ?? 'email'}
    />
  </Form.Group>
);

export default EmailField;
