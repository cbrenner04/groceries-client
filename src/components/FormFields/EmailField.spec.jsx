import React from 'react';
import { render, fireEvent } from '@testing-library/react';

import EmailField from './EmailField';

const defaultProps = {
  handleChange: jest.fn(),
  value: 'test@test',
};

describe('EmailField', () => {
  let formInput;

  beforeEach(() => {
    const { getByLabelText } = render(<EmailField {...defaultProps} />);
    formInput = getByLabelText('Email');
  });

  it('renders input', () => {
    const formGroup = formInput.parentElement;

    expect(formGroup).toMatchSnapshot();
    expect(formInput).toHaveValue(defaultProps.value);
  });

  describe('when value changes', () => {
    it('calls handleChange', () => {
      fireEvent.change(formInput, { target: { value: 'a' } });

      expect(defaultProps.handleChange).toHaveBeenCalled();
    });
  });
});
