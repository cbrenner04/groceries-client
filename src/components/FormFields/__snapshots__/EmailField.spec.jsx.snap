// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailField renders input 1`] = `
<div
  class="mb-3"
>
  <label
    class="form-label"
    for="email"
  >
    Email
  </label>
  <input
    class="form-control"
    id="email"
    name="email"
    placeholder="jane.smith@example.com"
    type="email"
    value="test@test"
  />
</div>
`;
